extensions/control-plane/provision/provision-bigquery/src/main/java/org/eclipse/edc/connector/provision/gcp/BigQueryConsumerResourceDefinitionGenerator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright (c) 2024 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LLC
+ *
+ */
+
+package org.eclipse.edc.connector.provision.gcp;
+
+import org.eclipse.edc.connector.transfer.spi.provision.ConsumerResourceDefinitionGenerator;
+import org.eclipse.edc.connector.transfer.spi.types.DataRequest;
+import org.eclipse.edc.connector.transfer.spi.types.ResourceDefinition;
+import org.eclipse.edc.gcp.bigquery.BigQueryService;
+import org.eclipse.edc.policy.model.Policy;
+import org.jetbrains.annotations.Nullable;
+
+import static java.util.UUID.randomUUID;
+
+public class BigQueryConsumerResourceDefinitionGenerator implements ConsumerResourceDefinitionGenerator {
+
+    @Override
+    public @Nullable
+    ResourceDefinition generate(DataRequest dataRequest, Policy policy) {
```

Comment:
heads up: the signature for this interface changed from `DataRequest -> TransferProcess`. you should rebase onto latest `main`

Reply:
Thank you, updated interface